unit-of-work/src/main/java/com/iluwatar/unitofwork/StudentRepository.java

```diff
@@ -0,0 +1,123 @@
+/*
+ * The MIT License (MIT)
+ *
+ * Copyright (c) 2014-2017 Piyush Chaudhari
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+package com.iluwatar.unitofwork;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * {@link StudentRepository} Student database repository.
+ * supports unit of work for student data.
+ */
+public class StudentRepository implements IUnitOfWork<Student> {
+  private static final Logger LOGGER = LoggerFactory.getLogger(StudentRepository.class);
+
+  private Map<String, List<Student>> context;
+  private StudentDatabase studentDatabase;
+
+  /**
+   * @param context         set of operations to be perform during commit.
+   * @param studentDatabase Database for student records.
+   */
+  public StudentRepository(Map<String, List<Student>> context, StudentDatabase studentDatabase) {
+    this.context = context;
+    this.studentDatabase = studentDatabase;
+  }
+
+  @Override
+  public void registerNew(Student student) {
+    LOGGER.info("Registering {} for insert in context.", student.getName());
+    register(student, IUnitOfWork.INSERT);
+  }
+
+  @Override
+  public void registerModified(Student student) {
+    LOGGER.info("Registering {} for modify in context.", student.getName());
+    register(student, IUnitOfWork.MODIFY);
+
+  }
+
+  @Override
+  public void registerDeleted(Student student) {
+    LOGGER.info("Registering {} for delete in context.", student.getName());
+    register(student, IUnitOfWork.DELETE);
+  }
+
+  private void register(Student student, String operation) {
+    List<Student> studentsToOperate = context.get(operation);
+    if (studentsToOperate == null) {
+      studentsToOperate = new ArrayList<>();
+    }
+    studentsToOperate.add(student);
+    context.put(operation, studentsToOperate);
+  }
+
+  @Override
+  public void commit() {
```

Comment:
I suggest we describe the behavior a bit, something like "Same type operations are batched and executed together on commit"

Reply:
@npathai : Totally agree. Added doc.